clients/src/main/java/org/apache/kafka/common/requests/admin/AlterTopicRequest.java

```diff
@@ -0,0 +1,153 @@
+
+            alterTopicRequestStructs.add(singleRequestStruct);
+        }
+        struct.set(REQUESTS_KEY_NAME, alterTopicRequestStructs.toArray());
+
+        this.alterTopicArguments = alterTopicArguments;
+    }
+
+    public AlterTopicRequest(Struct struct) {
+        super(struct);
+
+        Object[] requestStructs = struct.getArray(REQUESTS_KEY_NAME);
+        this.alterTopicArguments = new HashMap<String, AlterTopicArguments>();
+        for (Object requestStructObj : requestStructs) {
+            Struct singleRequestStruct = (Struct) requestStructObj;
+            String topic = singleRequestStruct.getString(TOPIC_KEY_NAME);
+            int partitions = singleRequestStruct.getInt(PARTITIONS_KEY_NAME);
+            int replicationFactor = singleRequestStruct.getInt(REPLICATION_FACTOR_KEY_NAME);
+
+            //replica assignment
+            Object[] assignmentsArray = singleRequestStruct.getArray(REPLICA_ASSIGNMENT_KEY_NAME);
+            List<PartitionReplicaAssignment> partitionReplicaAssignments = new ArrayList<PartitionReplicaAssignment>(assignmentsArray.length);
+            for (Object assignmentStructObj : assignmentsArray) {
+                Struct assignmentStruct = (Struct) assignmentStructObj;
+
+                Object[] replicasArray = assignmentStruct.getArray(REPLICA_ASSIGNMENT_REPLICAS_KEY_NAME);
+                List<Integer> replicas = new ArrayList<Integer>(replicasArray.length);
+                for (Object replica : replicasArray) {
+                    replicas.add((Integer) replica);
+                }
+                PartitionReplicaAssignment assignment = new PartitionReplicaAssignment(
+                        assignmentStruct.getInt(REPLICA_ASSIGNMENT_PARTITION_ID_KEY_NAME),
+                        replicas);
+
+                partitionReplicaAssignments.add(assignment);
+            }
+
+            AlterTopicArguments args = new AlterTopicArguments(
+                    partitions, replicationFactor, partitionReplicaAssignments);
+            alterTopicArguments.put(topic, args);
+        }
+    }
+
+    @Override
+    public AbstractRequestResponse getErrorResponse(int versionId, Throwable e) {
+        Map<String, Errors> topicErrors = new HashMap<String, Errors>();
+        for (String topic : alterTopicArguments.keySet()) {
+            topicErrors.put(topic, Errors.forException(e));
+        }
+
+        switch (versionId) {
+            case 0:
+                return new AlterTopicResponse(topicErrors);
+            default:
+                throw new IllegalArgumentException(String.format("Version %d is not valid. Valid versions for %s are 0 to %d",
+                        versionId, this.getClass().getSimpleName(), ProtoUtils.latestVersion(ApiKeys.ALTER_TOPIC.id)));
+        }
+    }
+
+
+    public Map<String, AlterTopicArguments> alterTopicArguments() {
+        return alterTopicArguments;
+    }
+
+    public static AlterTopicRequest parse(ByteBuffer buffer, int versionId) {
```

Comment:
strictly speaking, you don't need this yet. Still needed when we evolve I suppose